26-register_registration/2451-creator-adds-a-contributor.feature

```diff
@@ -0,0 +1,44 @@
+Feature: Creator adds a Contributor
+
+    Scenario: Creator opens the add Contributor Dialog
+        Given Creator navigates to Contributors tab
+        When they click "Add Contributor"
+        Then they see the "Add Contributor" Dialog
+        And they see a dropdown with Contributor types:
+            # Note: Contributor types that are required for the actual Registration
+            #       type can be omitted since they are covered by other Scenarios
```

Comment:
```suggestion
```

hvis det trengs kommentarer til et scenario, så er det kanskje noen steps som mangler eller?

Reply:
Dette er for å dekke feks Antologi, hvor `Editor` heller kan legges til i egen boks over med "Add Editor". Vet ikke hvordan jeg kan få med det uten å måtte lage eget scenario for alle typer, som virker tungvindt. Tror ikke det er noe i veien for at man legger til Editor her, så dukker den bare opp i lista over Editors i stedet for Contributors, så kan egentlig fjerne kommentaren ja(?)

Reply:
det er jo noe vi bør få avklart da. Kan du ha en Contributor med Contributor Type = Editor _OG_ en Editor? Er det det samme?

Reply:
Ja, det er det samme - derfor bør den undertrykkes når den er brukt som primær type (Creator)

Reply:
Hvis vi skal være mer spesifikke er kanskje dette bedre:
```
Scenario: Creator opens the Add Contributor Dialog
    Given Creator navigates to Contributors tab
    And the Registration is a "<RegistrationType>"
    When they click "Add Contributor"
    Then they see the "Add Contributor" Dialog
    And they see a dropdown with Contributor Types "<ContributorTypes>"
    And they see a "Close" Button
    And they see a "Create new Contributor" Button
    And they see a "Add me as Contributor" Button
    And they see a disabled "Add" Button
    Examples:
        | RegistrationType | ContributorTypes                                                                     |
        | BookAnthology   | Contact person,  Related person, Researcher, Rights holder, Supervisor, Other        |
        | Degree                | Contact person, Editor, Related person, Researcher, Rights holder,  Other            |
        | [Other]                | Contact person, Editor, Related person, Researcher, Rights holder, Supervisor, Other |
```
Da kan vi enkelt legge til andre typer for Forskningsdata senere også. Vet ikke hvordan vi skal få frem at det siste eksemplet er en hvilken som helst annen type. Alternativet hadde vært å liste opp alle subtyper, men det blir veldig mange eksempler igjen...

Reply:
yes, mye mer fan av eksplisitte examples! 👍

Reply:
hva syns du om eksempelet over @Eiriknil ?
merk at `BookAnthology` er subtype, `Degree` er egentlig alle subtypes av `Degree`, og `[Other]` er alle andre subtypes. mtp testing er det kanskje best å liste alle 18 subtypene selv om de fleste er like?

Reply:
Det er ein fordel om vi samlar denne kompleksiteten på ein story/senario. 
Det blir mange varianter under examples, men basis funksjonaliteten er den samme - uansett type ressurs.
Forskningsressurs vil vise variasjonen, men det er kunstnerisk utviklingsaktiveter som vil stå for den store variasjonen. Det er litt tid til vi skal starte på den delen av prosjektet, men det kjem.

Reply:
Jeg liker eksempelet over, Det vil gjøre det enklere å skrive gode tester.

Reply:
Og det er nok best å liste alle subtypene, ellers kommer testen til å bli noe kompleks.